src/components/SortPicker.tsx

```diff
@@ -0,0 +1,190 @@
+import useWindowOverflow from '../hooks/useWindowOverflow'
+import getSortPreference from '../selectors/getSortPreference'
+import rootedParentOf from '../selectors/rootedParentOf'
+import simplifyPath from '../selectors/simplifyPath'
+import fastClick from '../util/fastClick'
+import head from '../util/head'
+import isRoot from '../util/isRoot'
+import TriangleDown from './TriangleDown'
+
+// Define the type for sort options
+type SortType = 'None' | 'Alphabetical'
+
+// Default sort preference object
+const DEFAULT_SORT_PREFERENCE = { type: 'None', direction: null }
+
+/** Simple selector function to get sort preference. */
+const getSortPreferenceSelector = (state: State, cursor: Path | null) => {
+  if (!cursor || isRoot(cursor)) return DEFAULT_SORT_PREFERENCE
+
+  const path = rootedParentOf(state, cursor)
+  const simplePath = simplifyPath(state, path)
+  const id = head(simplePath)
+
+  return getSortPreference(state, id)
+}
+
+/** Sort Option component for individual sort options. */
+interface SortOptionProps {
+  type: SortType
+  isSelected: boolean
+  onClick: (type: SortType, e: React.MouseEvent | React.TouchEvent) => void
+  children: React.ReactNode
+}
+
+/** Render each sort option. */
+const SortOption: FC<SortOptionProps> = ({ type, isSelected, onClick, children }) => (
+  <div
+    key={type}
+    title={type}
+    className={css({
+      margin: '2px',
+      padding: '4px 8px',
+      lineHeight: '1.5',
+      display: 'flex',
+      flexDirection: 'row',
+      alignItems: 'center',
+      justifyContent: 'center',
+      border: isSelected ? `solid 1px {colors.fg}` : `solid 1px {colors.transparent}`,
+      cursor: 'pointer',
+      fontSize: '0.9em',
+      borderRadius: '2px',
+    })}
+    aria-label={type}
+    {...fastClick(e => e.stopPropagation())}
+    onTouchStart={e => onClick(type, e)}
+    onMouseDown={e => !isTouch && onClick(type, e)}
+  >
+    {children}
+  </div>
+)
+
+/** Sort Picker component. */
+const SortPicker: FC<{ fontSize: number; cssRaw?: SystemStyleObject }> = memo(({ fontSize, cssRaw }) => {
+  const ref = useRef<HTMLDivElement>(null)
+  const dispatch = useDispatch()
+  const overflow = useWindowOverflow(ref)
+
+  const cursor = useSelector(state => state.cursor)
+  const simplePath = useSelector(state =>
+    cursor && !isRoot(cursor) ? simplifyPath(state, rootedParentOf(state, cursor)) : null,
+  )
+
+  const sortPreference = useSelector(state => getSortPreferenceSelector(state, state.cursor), isEqual)
+
+  /** Toggles the sort option. */
+  const toggleSortOption = (
+    type: 'None' | 'Alphabetical' | 'Created' | 'Updated',
+    e: React.MouseEvent | React.TouchEvent,
+  ) => {
+    e.stopPropagation()
+    e.preventDefault()
+
+    if (!cursor || isRoot(cursor) || !simplePath) return
+
+    // Close the dropdown after selection
+    dispatch(toggleSortPicker({ value: false }))
+
+    // If clicking on 'None', reset to manual sorting
+    if (type === 'None') {
```

Comment:
All three of these branches end up calling `setSortPreference(simplePath, ...)`. Instead of separating them out like this, take advantage of inline ternary expressions to make a single call to `setSortPreference` with the appropriate values. 

The way this is written, it gives the impression that there are three different branches that do three totally different things, which is not the case. It obscures the similarity and increases the amount of code. In general you want to push conditions *down* as far as possible to consolidate shared functionality and decrease code duplication.

This structuring is indicative of thinking of code in terms of procedural steps (1, 2, 3, ...) rather than functions with a domain + range ([a1, a2] → [b1, b2, b3]).